app/models/user.rb

```
@@ -14,7 +14,7 @@ class User < ApplicationRecord
 
   validates :password, presence: true,
     length: {minimum: Settings.validates.password_min_length},
-    if: :password
+    allow_nil: true
```

allow_nil có tác dụng j

---

allow_nil dùng để bỏ qua validation khi giá trị là nil. Bởi mặc định khi gọi tới phương thức của đối tượng liên quan mà bị nil, lập tức sẽ báo lỗi "NoMethodError".Thay vì phải if else cho trường hợp này thì allow_nil sẽ giúp chúng ta.

---

create vs create! có gì khác nhau. tại sao trong file seed lại viết create! mà k phải create?
Trả lời: phương thức create! cũng giống như phương thức create, nhưng nó tạo ra một ngoại lệ cho một người dùng không hợp lệ thay vì trả về false. Hành vi này làm cho việc gỡ lỗi dễ dàng hơn bằng cách tránh các silent errors.
Lý do trong file seed lại dùng create!
Vì file seeds là file để tạo csdl với những người dùng mẫu theo 1 tiêu chuẩn được liệt kê trong file db/seeds.rb. Thế nên sẽ có những người dùng không hợp lệ trong quá trình tạo và vì không muốn các người dùng đó trả về false(nếu sử dụng phương thức create) thì phải sử dụng create!